lib/index.ts

```diff
@@ -0,0 +1,405 @@
+  total_votes: number;
+}
+
+export interface ServerCountHistoricalData extends BaseHistoricalData {
+  type: "server_count";
+  server_count: number;
+}
+
+export interface ShardCountHistoricalData extends BaseHistoricalData {
+  type: "shard_count";
+  shard_count: number;
+}
+
+export type HistoricalData =
+  | MonthlyVotesHistoricalData
+  | TotalVotesHistoricalData
+  | ServerCountHistoricalData
+  | ShardCountHistoricalData;
+
+export interface HistoricalDataResponse {
+  data: HistoricalData[];
+}
+
+export interface HistoricalDataRequest {
+  id: string;
+  timeFrame: HistoricalDataTimeFrame;
+  type: HistoricalDataTypes;
+}
+
+export interface HistoricalDataResponse {
+  data: HistoricalData[];
+}
+
+/**
+ * Structure for recent bot statistics
+ */
+export interface RecentData {
+  time: string;
+  monthly_votes: number;
+  server_count: number;
+  shard_count: number;
+  total_votes: number;
+  monthly_votes_change: number;
+  monthly_votes_change_perc: number;
+  server_count_change: number;
+  shard_count_change: number;
+  total_votes_change: number;
+}
+
+export interface RecentDataRequest {
+  id: string;
+}
+
+export interface RecentDataResponse {
+  hourlyData: RecentData[];
+  dailyData: RecentData[];
+}
+
+/**
+ * Branded type for rankings limit (1-500)
+ */
+export type RankingsLimit = number & { readonly __brand: unique symbol };
+
+/**
+ * Validates if a number is a valid rankings limit
+ */
+export function isValidRankingsLimit(value: number): value is RankingsLimit {
+  return Number.isInteger(value) && value >= 1 && value <= 500;
+}
+
+export const DEFAULT_RANKINGS_LIMIT = 100 as RankingsLimit;
+
+export type RankingsSortBy =
+  | "monthly_votes_rank"
+  | "total_votes_rank"
+  | "server_count_rank"
+  | "shard_count_rank";
+
+export type RankingsSortMethod = "asc" | "desc";
+
+export interface RankingsRequest {
+  limit?: RankingsLimit;
+  sortBy: RankingsSortBy;
+  sortMethod: RankingsSortMethod;
+}
+
+export interface RankingsData {
+  id: string;
+  name: string;
+  monthly_votes: number;
+  monthly_votes_rank: number;
+  server_count: number;
+  server_count_rank: number;
+  total_votes: number;
+  total_votes_rank: number;
+  shard_count: number;
+  shard_count_rank: number;
+  monthly_votes_rank_change: number;
+  server_count_rank_change: number;
+  total_votes_rank_change: number;
+  shard_count_rank_change: number;
+}
+
+export interface RankingsResponse {
+  totalBotCount: number;
+  data: RankingsData[];
+}
+
+/**
+ * Custom error classes for better error handling
+ */
+export class TopStatsError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "TopStatsError";
+    Object.setPrototypeOf(this, TopStatsError.prototype);
+  }
+}
+
+export class RateLimitError extends TopStatsError {
+  constructor(
+    message: string,
+    public expiresIn: string,
+  ) {
+    super(message);
+    this.name = "RateLimitError";
+    Object.setPrototypeOf(this, RateLimitError.prototype);
```

Comment:
What's this for?

Reply:
Setting the prototype here allowing for instanceof checks

```ts
try {
  throw new RateLimitError("Too many requests", 3600);
} catch (error) {
  if (error instanceof RateLimitError) {
    console.log(error.expiresIn);   // 3600
  }
}```